samcli/lib/providers/sam_base_provider.py

```diff
@@ -107,45 +128,6 @@ def _extract_sam_function_imageuri(resource_properties: Dict, code_property_key:
         """
         return resource_properties.get(code_property_key, None)
 
-    @staticmethod
-    def _extract_sam_function_codeuri(
-        name: str,
-        resource_properties: Dict[str, Any],
-        code_property_key: str,
-        ignore_code_extraction_warnings: bool = False,
-    ) -> str:
-        """
-        Extracts the SAM Function CodeUri from the Resource Properties
-
-        Parameters
-        ----------
-        name str
-            LogicalId of the resource
-        resource_properties dict
-            Dictionary representing the Properties of the Resource
-        code_property_key str
-            Property Key of the code on the Resource
-        ignore_code_extraction_warnings
-            Boolean to ignore log statements on code extraction from Resources.
-
-        Returns
-        -------
-        str
-            Representing the local code path
-        """
-        codeuri = resource_properties.get(code_property_key, SamBaseProvider.DEFAULT_CODEURI)
-        # CodeUri can be a dictionary of S3 Bucket/Key or a S3 URI, neither of which are supported
-        if isinstance(codeuri, dict) or (isinstance(codeuri, str) and codeuri.startswith("s3://")):
-            if not ignore_code_extraction_warnings:
-                LOG.warning(
-                    "Lambda function '%s' has specified S3 location for CodeUri which is unsupported. "
-                    "Using default value of '%s' instead",
-                    name,
-                    SamBaseProvider.DEFAULT_CODEURI,
-                )
-            return SamBaseProvider.DEFAULT_CODEURI
```

Comment:
explain: now the warning is being handled by `SamFunctionProvider` and `SamLayerProviders`.